ui/src/store/features/github/async-thunk/github-api.tsx

```diff
@@ -0,0 +1,76 @@
+import { ComboboxItem, DropdownItem } from '@/components';
+import { githubActions, RootState } from '@/store';
+import { useGithub } from '@/views/mint/github-step/steps/use-github';
+import { Repo } from '@/views/mint/mint.context';
+import { createAsyncThunk } from '@reduxjs/toolkit';
+
+export const fetchRepositoriesThunk = createAsyncThunk(
+  'github/fetchRepositories',
+  async (url: string, { dispatch, getState }) => {
+    debugger;
+    if ((getState() as RootState).github.queryLoading === 'loading') return;
+    const { fetchRepos } = useGithub();

```

Comment:
This seems circular to be a circular dependency. We should not have hooks inside an async thunk. Hooks should only be used for React components.